asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -47,114 +45,50 @@ def fetch_model_from_spec(
     if isinstance(models, str):
         models = [models]
 
+    if not local_dir:
+        raise ValueError("local_dir must be provided and must not be falsy.")
+
+    if not Path(local_dir).exists():
+        Path(local_dir).mkdir(parents=True, exist_ok=True)
+
     specs = {}
-    # cannot specify the same weights file for multiple models
-    weights_set = set()
     for model in models:
```

Comment:
I think we should add checks to make sure that all the necessary keys are present:
 - `model` in `spec`

 - "type" in `model_spec`
 - "base_url" in `model_spec`
 - "weights" in `model_spec`

 - "resource" in `weights`
 - "sha256hash" in `weights`

 - "resource" in `config`
 - "sha256hash" in `config`